examples/uexpr-tests.dx

```diff
@@ -258,3 +258,57 @@ def eitherFloor (x:(Int|Float)) : Int = case x of
 
 :p (eitherFloor (Left 1), eitherFloor (Right 2.3))
 > (1, 2)
+
+
+:p
+  [(a, b), c] = [(1, 2), (3, 4)]
+  (a, c)
+> (1, (3, 4))
+
+:p
+  [a, b] = [1, 2, 3]
+  (a, b)
+> Type error:Incorrect length of table pattern: table index set has 3 elements but there are 2 patterns.
+>
+>   [a, b] = [1, 2, 3]
+>   ^^^^^^^
+
+:p
+  [[a, _], [_, d]] = [[1, 2], [3, 4]] : ((Unit & Fin 2) => (Unit & Fin 2) => Int)
+  a + d
+> 5
+
+:p
+  [a, b, c, d] = [1, 2, 3, 4] : ((Fin 2 & Fin 2) => Int)
+  (a, b, c, d)
+> (1, (2, (3, 4)))
+
+
+:p [1, 2, 3, 4] : ((Fin _ & Fin 2) => Int)
+> Type error:
+> Expected: ((Fin a & Fin 2) => Int32)
+>   Actual: ((Fin 4) => Int32)
+> (Solving for: [a:Int32])
+> If attempting to construct a fixed-size table not indexed by 'Fin n' for some n, this error may indicate there was not enough information to infer a concrete index set; try adding an explicit annotation.
+>
+> :p [1, 2, 3, 4] : ((Fin _ & Fin 2) => Int)
+>    ^^^^^^^^^^^^^
+
+:p
+  x : ((Fin _ & Fin 2) => Int) = todo
+  [a, b, c, d] = x
+  (a, b, c, d)
+> Type error:
+> Expected: ((Fin a & Fin 2) => Int32)
+>   Actual: ((Fin 4) => b)
+> (Solving for: [a:Int32, b:Type])
+> If attempting to unpack a fixed-size table not indexed by 'Fin n' for some n, this error may indicate there was not enough information to infer a concrete index set; try adding an explicit annotation.
+>
+>   [a, b, c, d] = x
+>   ^^^^^^^^^^^^^
+
+:p
+  -- parentheses need to stop the parser from reading "zero" as a binder name
```

Comment:
Why so? It looks like a bug to me

Reply:
Without parentheses, this reads
```
[a, b, c, d] = zero : Fin _ => Int
```
which the parser interprets as a Pi type, binding a dependent argument called `zero` (and then ignoring it and producing an Int). That's also a valid interpretation of this expression, so I'm not sure how much we can do here.

It's not enough to parenthesize the RHS, either, because then it sees
```
zero : (Fin _ => Int)
```
and interprets it as the start of a Pi type, binding `zero` of type `Fin _ => Int` and then expecting an arrow. This is very strange behavior but it seems kind of out of scope to address here.

Reply:
lol, nice one. Yeah, this is out of scope, but it's an interesting failure

Reply:
Filed #282 to keep track of this, although it's probably low-priority since there's an easy workaround.

Reply:
The workaround is easy, but it's not necessarily easy to find, so that increases the priority a little 😛